### ISSUE TITLE
GHC 9.6 failures

### ISSUE THREAD

**Issue by locallycompact** [haskell-beam/beam#665]
```
beam-core> Database/Beam/Backend/SQL/Row.hs:195:7: error: [GHC-88464]
beam-core>     Variable not in scope:
beam-core>       replicateM_
beam-core>         :: Int -> FromBackendRowM be0 () -> FromBackendRowM be b0
beam-core>     Suggested fix:
beam-core>       Perhaps use one of these:
beam-core>         ‘Vector.replicateM’ (imported from Data.Vector.Sized),
beam-core>         ‘Vector.replicateM'’ (imported from Data.Vector.Sized),
beam-core>         ‘replicate’ (imported from Prelude)
beam-core>     |
beam-core> 195 |       replicateM_ (valuesNeeded (Proxy @be) (Proxy @(Maybe x)))
beam-core>     |       ^^^^^^^^^^^
beam-core> [12 of 32] Compiling Database.Beam.Backend.URI ( Database/Beam/Backend/URI.hs, dist/build/Database/Beam/Backend/URI.o, dist/build/Database/Beam/Backend/URI.dyn_o )
beam-core> [13 of 32] Compiling Database.Beam.Schema.Tables ( Database/Beam/Schema/Tables.hs, dist/build/Database/Beam/Schema/Tables.o, dist/build/Database/Beam/Schema/Tables.dyn_o )
beam-core> Database/Beam/Schema/Tables.hs:149:57: error: [GHC-76037]
beam-core>     Not in scope: type constructor or class ‘Endo’
beam-core>     |
beam-core> 149 | newtype EntityModification f be e = EntityModification (Endo (f e))
beam-core>     |                                                         ^^^^
error: builder for '/nix/store/99kzk7v85fc00smwq9ji7v47hrckbj28-beam-core-0.10.0.0.drv' failed with exit code 1;
       last 10 log lines:
       > 195 |       replicateM_ (valuesNeeded (Proxy @be) (Proxy @(Maybe x)))
       >     |       ^^^^^^^^^^^
       > [12 of 32] Compiling Database.Beam.Backend.URI ( Database/Beam/Backend/URI.hs, dist/build/Database/Beam/Backend/URI.o, dist/build/Database/Beam/Backend/URI.dyn_o )
       > [13 of 32] Compiling Database.Beam.Schema.Tables ( Database/Beam/Schema/Tables.hs, dist/build/Database/Beam/Schema/Tables.o, dist/build/Database/Beam/Schema/Tables.dyn_o )
       >
       > Database/Beam/Schema/Tables.hs:149:57: error: [GHC-76037]
       >     Not in scope: type constructor or class ‘Endo’
       >     |
       > 149 | newtype EntityModification f be e = EntityModification (Endo (f e))
       >     |                                                         ^^^^
       For full logs, run 'nix log /nix/store/99kzk7v85fc00smwq9ji7v47hrckbj28-beam-core-0.10.0.0.drv'.
```

source: https://gitlab.horizon-haskell.net/package-sets/horizon-platform/-/jobs/73022

**Comment by tathougies** [comment]
Thanks for the report... It looks like these are mainly due to movement of definitions. Will have a PR up in the next few days.

**Comment by maerten** [comment]
This fork by @mightybyte compiles for me on GHC9.6.2 (MacOS)

https://github.com/mightybyte/beam/commit/7bf5d81d358063a0106df61b529b4742f43fb816

You can add the fork and fix to extra-deps in stack.yaml:

```
extra-deps:
- github: mightybyte/beam
  commit: 7bf5d81 # commit that contains the fix
  subdirs:
   - beam-core
   - beam-postgres
   - beam-migrate
```

Would be great if a PR would be submitted for it though!

**Comment by kmicklas** [comment]
The specific error here is fixed by #673.
